onnxruntime/core/providers/rocm/tunable/gemm_hipblaslt.h

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifdef USE_HIPBLASLT
+#include <hipblaslt/hipblaslt.h>
+#endif
+
+#include "core/common/common.h"
+#include "core/providers/rocm/tunable/gemm_common.h"
+#include "core/providers/rocm/tunable/rocm_tunable.h"
+
+namespace onnxruntime {
+namespace rocm {
+namespace tunable {
+namespace blas {
+namespace internal {
+
+#ifdef USE_HIPBLASLT
+
+template <typename T>
+constexpr hipblasDatatype_t HipBlasDataTypeFor(const T*) {
+  static_assert(sizeof(T) == 0, "Unsupported data type for hipBLASLt operation.");
+  // Compiler will complain if we don't return something.
+  return HIPBLAS_R_32F;
+}
```

Comment:
I don't believe so, how about change it to
```cpp
template <typename T>
constexpr hipblasDatatype_t HipBlasDataTypeFor(const T*);
```

Reply:
Here I wanted to have the error message thrown for unsupported cases. With this `static_assert` it will need a return, right?

Reply:
@cloudhan's suggestion will also report an error message for unsupported cases, and it's fewer lines of code :-)